Bugfix: Fix crashes

wutschel opened the pull request:
## Description
<!--- Detailed info for reviewers and developers -->
Closes https://github.com/xbmc/Official-Kodi-Remote-iOS/issues/604.
Closes https://github.com/xbmc/Official-Kodi-Remote-iOS/issues/606.

1. Out-of-bounds crash in iPad's `StackViewController`. In 1.10 test builds this could be triggered by pressing remote's settings twice.
2. Crash of `componentsWithURL:resolvingAgainstBaseURL:` when handing `url == nil`.
3. Crash of `checkDiskCache` after wakeup.
3. Adding a "close" button to the iPad's remote toolbar to allow closing the modal view.

## Summary for release notes
<!--- This will be shown in Testflight to end users / testers -->
<!--- Please fill in, as usually PR title isn't clear to ordinary users -->
<!--- If your changes don't modify app files, please add prefix [not app] -->
Bugfix: Fix crash when loading movie details (processing trailer URL)
Bugfix: Fix crash when pressing remote setting twice on iPad (1.10 test builds only)
Bugfix: Fix possible UI lock up when entering remote on iPad and iOS 13 and earlier (1.10 test builds only)

kambala-decapitator reviewed (approved):
I see that in 98c2370203a9eb79a93603e85149ea6b0a091efe indentation is broken, probably a mix of tabs and spaces

wutschel commented:
> I see that in [98c2370](https://github.com/xbmc/Official-Kodi-Remote-iOS/commit/98c2370203a9eb79a93603e85149ea6b0a091efe) indentation is broken, probably a mix of tabs and spaces

Yes, in this file this is the case in many places. I added a fixup to unify the indentation around the code I changed. Shall I squash it with the code change itself?

kambala-decapitator commented:
> Shall I squash it with the code change itself?

yes please

wutschel commented:
Squashed and rebased to master.
